code/modules/mob/living/carbon/xenomorph/castes/spiderling/spiderling.dm

```diff
@@ -44,14 +44,13 @@
 	RegisterSignal(escorted_atom, COMSIG_XENOMORPH_RESIN_JELLY_APPLIED, PROC_REF(apply_spiderling_jelly))
 	RegisterSignals(escorted_atom, list(COMSIG_XENOMORPH_REST, COMSIG_XENOMORPH_UNREST), PROC_REF(toggle_rest))
 	RegisterSignal(escorted_atom, COMSIG_SPIDERLING_MARK, PROC_REF(decide_mark))
+	RegisterSignal(escorted_atom, COMSIG_SPIDERLING_RETURN, PROC_REF(only_set_escorted_atom))
 
 /// Decides what to do when widow uses spiderling mark ability
 /datum/ai_behavior/spiderling/proc/decide_mark(source, atom/A)
 	SIGNAL_HANDLER
 	if(!A)
 		only_set_escorted_atom()
-		UnregisterSignal(atom_to_walk_to, list(COMSIG_MOB_DEATH, COMSIG_QDELETING))
```

Comment:
You sure removing this won't cause issues?

Reply:
https://www.youtube.com/watch?v=SM69WsJ8r9k

Reply:
Guess we'll see in the TM. Also don't self-resolve stuff that isn't resolved.

Reply:
woops! got it.